mpa/modules/hooks/checkpoint_hook.py

```diff
@@ -53,14 +53,15 @@ def after_train_epoch(self, runner):
             return
         if hasattr(runner, 'save_ckpt'):
             if runner.save_ckpt:
-                if runner.save_ema_model:
+                # TODO: Do we need this even after we moved from torchreid to mmcls?
+                if getattr(runner, "save_ema_model", False):
                     backup_model = runner.model
                     runner.model = runner.ema_model
                 runner.logger.info(f'Saving checkpoint at {runner.epoch + 1} epochs')
                 if self.sync_buffer:
                     allreduce_params(runner.model.buffers())
                 self._save_checkpoint(runner)
-                if runner.save_ema_model:
+                if getattr(runner, "save_ema_model", False):
```

Comment:
Normally, runner should have this argument unless someone adds a runner, which is not inherited from `EpochRunnerWithCancel`, so this workaround is not required.

Reply:
`AccuracyAwareRunner` which is a runner for `NNCF` task is not inherited from `EpochRunnerWithCancel`.

Reply:
Then it's a violation of LSP, the inheritance hierarchy is broken. If we have a classification-specific implementation of a NNCF task, why can't we make a subclass inherited from both this runner and AccuracyAwareRunner?

Reply:
Do you mean `*RunnerWithCancel` should **the super classes** not `*Runner` from mmcv in OTX? If it is, I would just make `AccuracyAwareRunner` be inherited from `EpochRunnerWithCancel`.

Reply:
Yes, that's it. All cls runners are supposed to implement functionality from `EpochRunnerWithCancel`

Reply:
Fixed in https://github.com/openvinotoolkit/model_preparation_algorithm/pull/106/commits/1bee390495bce892270de1be3bc286b6c42fce85